### DISCUSSION TITLE
Remove password protection on PDF files upon upload

### DISCUSSION THREAD

**Discussion by ronnyandre** [jonaswinkler/paperless-ng#812] [Feature Requests]
I upload my montly paychecks for historical reasons, but these are password protected from my employer's HR system. It would be nice if it was possible to remove this password protection as the password is different from employer to employer, giving me three different passwords from the past 10 years.

**Comment by T-x-T** [comment]
I would love to see this too, as I have exactly the same issue. This could perhaps be an option in the automated email import

**Comment by jonaswinkler** [comment]
I haven't tried that myself yet, but I believe you should be able to write a pre-consume script for that. The pre consume script is a shell script that receives filenames of documents that are about to get consumed. Replace the file in place with a decrypted version, and you're good to go.
